crates/biome_css_analyze/src/lint/nursery/no_font_family_duplicate_names.rs

```diff
@@ -0,0 +1,200 @@
+use std::collections::HashSet;
+
+use biome_analyze::{context::RuleContext, declare_rule, Ast, Rule, RuleDiagnostic, RuleSource};
+use biome_console::markup;
+use biome_css_syntax::{
+    AnyCssGenericComponentValue, AnyCssValue, CssGenericComponentValueList, CssGenericProperty,
+};
+use biome_rowan::{AstNode, SyntaxNodeCast, TextRange};
+
+use crate::utils::{is_css_variable, is_font_family_keyword, is_font_shorthand_keyword};
+
+declare_rule! {
+    /// Disallow duplicate names within font families.
+    ///
+    /// This rule checks the `font` and `font-family` properties for duplicate font names.
+    ///
+    /// This rule ignores var(--custom-property) variable syntaxes.
+    ///
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```css,expect_diagnostic
+    /// a { font-family: "Lucida Grande", 'Arial', sans-serif, sans-serif; }
+    /// ```
+    ///
+    /// ```css,expect_diagnostic
+    /// a { font-family: 'Arial', "Lucida Grande", Arial, sans-serif; }
+    /// ```
+    ///
+    /// ```css,expect_diagnostic
+    /// a { FONT: italic 300 16px/30px Arial, " Arial", serif; }
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```css
+    /// a { font-family: "Lucida Grande", "Arial", sans-serif; }
+    /// ```
+    ///
+    /// ```css
+    /// b { font: normal 14px/32px -apple-system, BlinkMacSystemFont, sans-serif; }
+    /// ```
+    pub NoFontFamilyDuplicateNames {
+        version: "next",
+        name: "noFontFamilyDuplicateNames",
```

Comment:
What do you think about `noRedundantFontName`? We already have some rules that follow this naming convention

Reply:
Or `noDuplicateFontNames` to align with our existing rule names?